hippoclient/textedit.py

```diff
@@ -0,0 +1,462 @@
+                                tooltip="Enter the absolute filepath to a file of the same name to update the file; leave this field as-is if only editing other fields (i.e., source description)",
+                            )
+                        with Vertical():
+                            yield Label("Source Description:", variant="accent")
+                            yield Input(
+                                value=source.description,
+                                placeholder="Description...",
+                                id=f"edit-source-desc-{source.uuid}",
+                            )
+                        with Vertical():
+                            yield Label("Is Primary Source?", variant="accent")
+                            yield Checkbox(
+                                "Primary",
+                                classes="primary-source-checkbox",
+                                id=f"edit-primary-source-checkbox-{source.uuid}",
+                            )
+                        with Vertical():
+                            yield Label("Delete Source?", variant="accent")
+                            yield Checkbox(
+                                "Yes",
+                                classes="delete-source-checkbox",
+                                id=f"delete_source_checkbox_{source.uuid}",
+                            )
+                with Horizontal(classes="sources new-source"):
+                    with Vertical():
+                        yield Label("Filepath:", variant="accent")
+                        yield Input(
+                            value=None,
+                            placeholder="Path to new source file...",
+                            id="new-source-file-path",
+                        )
+                    with Vertical():
+                        yield Label("Source Description:", variant="accent")
+                        yield Input(
+                            value=None,
+                            placeholder="Source description...",
+                            id="new-source-file-desc",
+                        )
+                    with Vertical():
+                        yield Label("Is Primary Source?", variant="accent")
+                        yield Checkbox(
+                            "Primary",
+                            classes="primary-source-checkbox",
+                            id="new-primary-source-checkbox",
+                        )
+                with Horizontal(classes="save-or-cancel"):
+                    yield Button("Save", variant="success", id="save")
+                    yield Button("Cancel", variant="error", id="cancel")
+
+            with TabPane("Metadata", id="metadata-tab"):
+                class_schema = self._metadata.model_json_schema()
+                class_title = class_schema["title"]
+                yield Label("Metadata Type:", variant="primary")
+                metadata_options = []
```

Comment:
metadata_options = [v.model_json_schema()["title"] for v in ALL_METADATA.values() if v is not None]